src/java/org/apache/cassandra/index/sai/disk/v1/VectorIndexSearcher.java

```diff
@@ -233,7 +235,9 @@ public RangeIterator<PrimaryKey> limitToTopResults(QueryContext context, RangeIt
 
             // else ask hnsw to perform a search limited to the bits we created
             float[] queryVector = exp.lower.value.vector;
-            var results = graph.search(queryVector, limit, bits, Integer.MAX_VALUE, context);
+            // We have ss table row ids, so we set the RowIdToPrimaryKeyMapper to use a 0 offset
```

Comment:
The argument is not always a segment row id. The `limitToTopResults` method that we're in here gets an iterator of the SSTable row ids. The `RowIdToPrimaryKeyMapper` class takes as an argument an offset so that it can work with either segment row ids or SSTable row ids. Perhaps this design isn't right, but I'm pretty sure I am right about the ids we are iterating over.

Reply:
In https://github.com/datastax/cassandra/commit/65a678a34d99a465926b92947c936eedc2d34eb8#diff-c4f269386c2a2e56f94d52cbba5e0eaa24d16f5b6bdd6c2679632bf998c89e13R730 hybrid search test:
- Populated a sstable with 3 segments, each segment has 1 row. Their segment ids are all 0.
- In order to test hybrid search with small data size, brute force ANN search is disabled.

The test log shows that:
* It's segment row id in `getPrimaryKeyForRowId()` argument
* Score is -1 for all 3 rows because segment offset wasn't used and they get the same primary-key

Reply:
> The limitToTopResults method that we're in here gets an iterator of the SSTable row ids. 

the HNSW on disk structure stores segment row ids..

Reply:
Thanks, I see what you're saying now. I think was confused because the iterator has SSTable Row Ids, but the bits that we're limiting in the HNSW search are based on the segment ids.

Reply:
I pushed up a fix. PTAL.